src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtil.java

```diff
@@ -353,4 +353,33 @@ public static BitSet asBitSet(String... tokens) {
                 .collect(BitSet::new, BitSet::set, BitSet::or);
     }
 
+    /**
+     * Checks whether the specified type has all the specified modifiers.
+     *
+     * @param ast the AST node to check
+     * @param modifiersTypes int array of all modifiers types to check.
+     * @return true if AST node has all specified modifiers.
+     */
+    public static boolean hasModifiers(DetailAST ast, int... modifiersTypes) {
```

Comment:
Pointing to https://github.com/checkstyle/checkstyle/issues/11527 over past concerns with performance with these type of methods.